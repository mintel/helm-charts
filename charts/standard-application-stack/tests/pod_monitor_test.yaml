suite: Test PodMonitors
templates:
  - pod-monitor.yaml
release:
  namespace: test-namespace
tests:
  - it: Check name, namespace and labels
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: false
      metrics.enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: PodMonitor
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: test-app
      - equal:
          path: metadata.labels.[app.kubernetes.io/name]
          value: test-app
      - equal:
          path: metadata.labels.[name]
          value: test-app
  - it: Check default endpoints
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: false
      metrics.enabled: true
    asserts:
      - isKind:
          of: PodMonitor
      - equal:
          path: spec.podMetricsEndpoints
          value:
            - interval: 30s
              path: /metrics
              port: http
              relabelings:
              - separator: /
                sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
                targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
  - it: Check metric overrides
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: false
      metrics.enabled: true
      metrics.path: /updated-metrics-path
      metrics.port: another-named-port
      metrics.scheme: HTTPS
      metrics.timeout: 20s
    asserts:
      - isKind:
          of: PodMonitor
      - equal:
          path: spec.podMetricsEndpoints
          value:
            - interval: 30s
              path: /updated-metrics-path
              port: another-named-port
              relabelings:
              - separator: /
                sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
                targetLabel: job
              scheme: HTTPS
              scrapeTimeout: 20s
              tlsConfig:
                insecureSkipVerify: true

  - it: Check basic-auth
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: false
      metrics.enabled: true
      metrics.basicAuth.enabled: true
      metrics.basicAuth.usernameKey: user
      metrics.basicAuth.passwordKey: passwd
    asserts:
      - isKind:
          of: PodMonitor
      - equal:
          path: spec.podMetricsEndpoints
          value:
            - interval: 30s
              path: /metrics
              port: http
              basicAuth:
                username:
                  key: user
                  name: test-app-app
                password:
                  key:  passwd
                  name: test-app-app
              relabelings:
              - separator: /
                sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
                targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
  - it: Check additional PodMonitor resources
    set:
      global.name: test-app
      global.clusterEnv: prod
      service.enabled: false
      metrics.enabled: true
      metrics.additionalMonitors:
        - name: monitor-1
          path: /metrics-1
        - name: monitor-2
          path: /metrics-2
          interval: 20s
          scheme: https
    asserts:
      - hasDocuments:
          count: 3
      - isKind:
          of: PodMonitor
        documentIndex: 0
      - equal:
          path: metadata.name
          value: test-app
        documentIndex: 0
      - isKind:
          of: PodMonitor
        documentIndex: 1
      - equal:
          path: metadata.name
          value: test-app-monitor-1
        documentIndex: 1
      - isKind:
          of: PodMonitor
        documentIndex: 2
      - equal:
          path: metadata.name
          value: test-app-monitor-2
        documentIndex: 2
      - equal:
          path: spec.podMetricsEndpoints
          value:
            - interval: 30s
              path: /metrics-1
              # FIXME: This should match main-monitor-1
              port: http
              relabelings:
              - separator: /
                sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
                targetLabel: job
              scheme: http
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 1
      - equal:
          path: spec.podMetricsEndpoints
          value:
            - interval: 20s
              path: /metrics-2
              # FIXME: This should match main-monitor-2
              port: http
              relabelings:
              - separator: /
                sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
                targetLabel: job
              scheme: https
              scrapeTimeout: 10s
              tlsConfig:
                insecureSkipVerify: true
        documentIndex: 2
