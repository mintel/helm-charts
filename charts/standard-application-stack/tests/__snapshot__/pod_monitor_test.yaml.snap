Check additional PodMonitor resources:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
  2: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-2
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-2
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-metrics-2
      name: test-app-metrics-2
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics-2
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-2
          app.kubernetes.io/name: test-app-metrics-2
  3: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-3
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-3
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-metrics-3
      name: test-app-metrics-3
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 20s
          path: /metrics-3
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: https
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-3
          app.kubernetes.io/name: test-app-metrics-3
Check basic-auth:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - basicAuth:
            password:
              key: passwd
              name: test-app-app
            username:
              key: user
              name: test-app-app
          interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
Check combined template defaults:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/env: prod
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: prod
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:v0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                apparmor:
                  enabled: false
                  profile: unconfined
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            allowPrivilegeEscalation: false
            apparmor:
              enabled: false
              profile: unconfined
            capabilities:
              drop:
                - ALL
            runAsNonRoot: true
            runAsUser: 1000
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: DoNotSchedule
  2: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
Check combined template with extra ports and monitors:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: test-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations: null
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            app.mintel.com/env: prod
            app.mintel.com/region: ${CLUSTER_REGION}
            name: test-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: prod
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: test-app-app
              image: registry.gitlab.com/test:v0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
                - containerPort: 9002
                  name: metrics-2
                - containerPort: 9003
                  name: metrics-3
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                apparmor:
                  enabled: false
                  profile: unconfined
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            allowPrivilegeEscalation: false
            apparmor:
              enabled: false
              profile: unconfined
            capabilities:
              drop:
                - ALL
            runAsNonRoot: true
            runAsUser: 1000
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: test-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: test-app
              maxSkew: 1
              topologyKey: kubernetes.io/hostname
              whenUnsatisfiable: DoNotSchedule
  2: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
  3: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-2
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-2
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-metrics-2
      name: test-app-metrics-2
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics-2
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-2
          app.kubernetes.io/name: test-app-metrics-2
  4: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: metrics-3
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app-metrics-3
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app-metrics-3
      name: test-app-metrics-3
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 20s
          path: /metrics-3
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: https
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: metrics-3
          app.kubernetes.io/name: test-app-metrics-3
Check default endpoints:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
Check default targetLabels:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
Check metric overrides:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /updated-metrics-path
          port: another-named-port
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: HTTPS
          scrapeTimeout: 20s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
Check name, namespace and labels:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: PodMonitor
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        app.mintel.com/env: prod
        app.mintel.com/region: ${CLUSTER_REGION}
        name: test-app
      name: test-app
      namespace: test-namespace
    spec:
      podMetricsEndpoints:
        - interval: 30s
          path: /metrics
          port: http
          relabelings:
            - separator: /
              sourceLabels:
                - __meta_kubernetes_namespace
                - __meta_kubernetes_pod_label_app_kubernetes_io_name
              targetLabel: job
          scheme: http
          scrapeTimeout: 10s
          tlsConfig:
            insecureSkipVerify: true
      podTargetLabels:
        - app.kubernetes.io/name
        - app.kubernetes.io/part-of
        - app.kubernetes.io/component
        - app.kubernetes.io/instance
        - app.mintel.com/owner
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: test-app
